### DISCUSSION TITLE
Feature request: Add styles for TreeDataGrid

### DISCUSSION THREAD

**Discussion by sn4k3** [kikipoulet/SukiUI#352] [Ideas]
Is it possible to add styles like what DataGrid has but to [TreeDataGrid](https://docs.avaloniaui.net/docs/reference/controls/treedatagrid/) ?

**Comment by sirdoombox** [comment]
I detailed why this is unlikely to happen here - #320, but generally speaking it's not worth the time and effort to support a control that Avalonia themselves don't support.

Avalonia actually recommends 3rd party implementations of DataGrid at this point over TreeDataGrid, which really says it all.

**Reply by kikipoulet** [reply]
If someone suggest making our own implementation, I scream to death 👀

**Reply by sn4k3** [reply]
I understand.
BTW is there any valid alternative for the Avalonia DataGrid?
The reason I gone TreeDataGrid is only because the annoyance of DataGrid when you change the ItemSource the sort from user is gone, and in my usage the ItemSource changes very often. By code I can restore the column sort but not the direction as seems there are no way to obtain the direction even when subscribe Sorting event.

**Reply by kikipoulet** [reply]
> I understand.
> BTW is there any valid alternative for the Avalonia DataGrid?
> The reason I gone TreeDataGrid is only because the annoyance of DataGrid when you change the ItemSource the sort from user is gone, and in my usage the ItemSource changes very often. By code I can restore the column sort but not the direction as seems there are no way to obtain the direction even when subscribe Sorting event.

I don't think there is a great DataGrid solution right now in Avalonia unfortunately.

However, I suppose it is just the styling of SukiUI DataGrid that you want for TreeDataGrid to keep the design consistent ?

**Reply by sn4k3** [reply]
> However, I suppose it is just the styling of SukiUI DataGrid that you want for TreeDataGrid to keep the design consistent ?

Correct

**Reply by sirdoombox** [reply]
https://www.eremexcontrols.com/ implementation is what I've seen mentioned and recommended.

Outside of paid solutions, there clearly hasn't been enough of an appetite for it for anyone to build one.

**Reply by sn4k3** [reply]
> https://www.eremexcontrols.com/ implementation is what I've seen mentioned and recommended.
> 
> Outside of paid solutions, there clearly hasn't been enough of an appetite for it for anyone to build one.

Looks exceptionally good, but too expensive model for an open-source app :)

BTW not related to this but I faced this alignment "problem":

![image](https://github.com/user-attachments/assets/bff5b0c4-7849-4caa-a60d-6435272c259c)

It is by design to have that margin/padding on the message content? Using Suki Dialog...

**Reply by kikipoulet** [reply]
I made a quick copycat style for treedatagrid :

![treedatagrid](https://github.com/user-attachments/assets/a85c2488-9c27-4694-8361-ba002854c5e5)

Just use this style file 
[TreeDataGridStyles.zip](https://github.com/user-attachments/files/18108805/TreeDataGridStyles.zip)

**Reply by kikipoulet** [reply]
Moreover, working quickly on TreeDataGrid made me realize how unnecesserily complicated this control is to use for the dev (Styling is easy however). We definitely don't want to support that kind of thing in an extensive way. Fortunately, the SukiUI library isn't really meant for accounting or ERP software.

However, on the opposite side, there isn't an easy solution to display nice Grid data, and maybe it can enter in our scope to write a 'wrapper' for TreeDataGrid with Column auto-generation to present list data nicely automatically. Maybe by using some logic of the PropertyGrid ?

A UI-first oriented grid alternative, something like that :

![Sans titre](https://github.com/user-attachments/assets/afc8e77d-a5bb-4b02-a307-b1d795131723)

Auto-generating columns with reflection and using different nice datatemplates for enum, percentage, etc .. Automatically create buttons for commands, etc ..

Just a thought that I had 🧐

**Reply by sn4k3** [reply]
> Just use this style file
[TreeDataGridStyles.zip](https://github.com/user-attachments/files/18108805/TreeDataGridStyles.zip)

Thank you! It now integrates better in the UI :)

> TreeDataGrid made me realize how unnecesserily complicated this control is to use

For flat data I find it easy, but yeah no bindings and many stuff missing from normal DataGrid is a fuss.
I post my issue with DataGrid [here](https://github.com/AvaloniaUI/Avalonia/discussions/17746)

> maybe it can enter in our scope to write a 'wrapper' for TreeDataGrid

Your library is one of the best I have seen, I know is a lot of work but if you can pull that out you will earn a lot of costumers!
Also I think `TreeDataGrid` is still developed, in a slow pace...

**Reply by sn4k3** [reply]
Also I found that `DataGrid` lacks the styles for groups, below I have DNS and Microsoft groups

```
var groupView = new DataGridCollectionView(_model.Hosts);
groupView.GroupDescriptions.Add(new DataGridPathGroupDescription("Group"));
PingHostsGrid.ItemsSource = groupView;
```

![image](https://github.com/user-attachments/assets/b2dcc291-f95d-40cf-8199-90cc17815923)
